src/checks.ts

```diff
@@ -145,3 +145,41 @@ export function noConsoleLogs (file: FileInformation): Discrepancy | undefined {
     }
   }
 }
+
+/**
+ * Checks to ensure there are no string concatenations written in the form "" + "" or '' + ''
+ * @param file - Information for the provided file
+ * @returns - A discrepancy with more information, or nothing if it does not break the rule
+ */
+export function dollarSignForStrConcat (file: FileInformation): Discrepancy | undefined {
+  const lines: string[] = file.content.replace(/\r\n/g,'\n').split('\n')
+  for (let lineIndex = 0; lineIndex < lines.length; lineIndex++) {
+    if (lines[lineIndex].includes('\' + \'') || lines[lineIndex].includes('\" + \"')) {
```

Comment:
This looks good! Here's a potential problem though, what would happen if I used the following line in some of my code:

```TS
const x: string = "Hello"+"World"
```

Would this still be picked up by the test? Is there a more generic way for us to determine if string concatenation has been used without caring about whitespace (i.e. the empty bits between " + ")?

<details><summary>Hint if you get stuck</summary>

If we remove all of the whitespace before we check the content of the line, it should _always_ be in the format of either "+" or '+'!

</details>